src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -262,7 +274,16 @@
           :keylength
           :manage-internal-file-permissions
           :ruby-load-path
-          :gem-path))
+          :gem-path
+          :compile-masters
+          ;; We plan to use one of the existing directories for maintaining the
+          ;; compile master serials and infra-crl so removing them from here
```

Comment:
I don't follow.

Reply:
So I ran into an issue. I removed the compile-master-serials-path and infra-crl-path from settings->cadir-paths. Then launched puppetserver while specifying these two configs to point to some non-existing paths. Would have expected the directories, from the paths specified in values of these two to get created during init (by create-parent-directories).
But it doesn't seem to. 
If I leave the above two as defaults then they end up re-using the same folder as that of cacrl (which does get created by create-parent-directories) and things work out then.

Reply:
Figured this